api/provider/aws/service_create.go

```diff
@@ -5,22 +5,66 @@ import (
 
 	"github.com/aws/aws-sdk-go/aws"
 	"github.com/aws/aws-sdk-go/service/ecs"
+	"github.com/quintilesims/layer0/common/config"
 	"github.com/quintilesims/layer0/common/models"
 )
 
+// Create runs an ECS Service using the specified CreateServiceRequest.
+// The CreateServiceRequest contains the DeployID, the EnvironmentID,
+// the LoadBalancerID (optional), the Scale, the ServiceName, and the ServiceType.
+//
+// The DeployID is used to look up the ARN of the ECS TaskDefinition to run. If a
+// LoadbalancerID is supplied, it will be used in conjunction with the TaskDefinition
+// ARN to compare the ports specified in the TaskDefinition with those specified on
+// the LoadBalancer. The ServiceType parameter is one of "stateful" or "stateless"
+// and indicates which ECS LaunchType the user wishes to use ("EC2" or "FARGATE"
+// respectively).
+//
+// Create does not generate any custom errors of its own, but will bubble up errors
+// found in its helper functions as well as errors returned by AWS.
 func (s *ServiceProvider) Create(req models.CreateServiceRequest) (string, error) {
 	fqEnvironmentID := addLayer0Prefix(s.Config.Instance(), req.EnvironmentID)
 	cluster := fqEnvironmentID
-
 	serviceID := entityIDGenerator(req.ServiceName)
 	fqServiceID := addLayer0Prefix(s.Config.Instance(), serviceID)
 	serviceName := fqServiceID
 
+	scale := req.Scale
+	if req.Scale == 0 {
```

Comment:
this should be handled by `models.CreateServiceRequest.Validate()`

Reply:
✅